src/components/__stories__/ColumnPinningDemo.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import type {ColumnPinningState} from '@tanstack/react-table';
+
+import {useTable} from '../../hooks';
+import {Table} from '../Table';
+
+import {columnPinningDemoBlock} from './ColumnPinningDemo.classname';
+import {data} from './constants/data';
+import {columns} from './constants/pinning';
+
+import './ColumnPinningDemo.scss';
+
+export const ColumnPinningDemo = () => {
```

Comment:
Please add another example/demo with rowSelection/reordering to check if these features work correctly together.